leval/evaluator.py

```diff
@@ -107,9 +118,18 @@ def visit(self, node):  # noqa: D102
     def visit_Compare(self, node):  # noqa: D102
         if len(node.ops) != 1:
             raise InvalidOperation("Only simple comparisons are supported", node=node)
-        left = self.visit(node.left)
-        right = self.visit(node.comparators[0])
         op = node.ops[0]
+        if type(op) in [ast.Is, ast.IsNot]:
```

Comment:
1. Use a tuple for `in` checks – Python will be able to optimize it somewhat more during compile
2. Use `isinstance` to check for types:

```suggestion
        if isinstance(op, (ast.Is, ast.IsNot)):
```